src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs

```diff
@@ -542,7 +542,13 @@ public enum SyntaxKind : ushort
         TupleType = 8925,
         TupleElement = 8926,
         TupleExpression = 8927,
-        VariableDeconstructionDeclarator = 8928,
+        TypedVariableComponent = 8928,
+        TupleDeconstructionVariableComponent = 8929,
+        SingleVariableDesignation = 8930,
+        TupleDeconstructionVariableDesignation = 8931,
+        DeconstructionDeclarationStatement = 8932,
+        VariableComponentAssignment = 8933,
+        ForEachComponentStatement = 8934,
 
         // patterns (for pattern-matching)
         DeclarationPattern = 9000,
```

Comment:
I understand that these numbers aren't public, but should we compact here too? This is an aside from this PR.